actors/migration/nv3/miner.go

```diff
@@ -1,30 +1,472 @@
 package nv3
 
 import (
+	"bytes"
 	"context"
-
-	"github.com/filecoin-project/go-address"
+	addr "github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-bitfield"
 	"github.com/filecoin-project/go-state-types/abi"
 	cid "github.com/ipfs/go-cid"
 	cbor "github.com/ipfs/go-ipld-cbor"
+	"golang.org/x/xerrors"
 
+	"github.com/filecoin-project/specs-actors/actors/builtin"
 	miner "github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/actors/builtin/power"
 	"github.com/filecoin-project/specs-actors/actors/states"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
 )
 
 type minerMigrator struct {
 }
 
-func (m *minerMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, head cid.Cid, _ abi.ChainEpoch, _ address.Address, _ *states.Tree) (cid.Cid, error) {
+func (m *minerMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, head cid.Cid, priorEpoch abi.ChainEpoch, a addr.Address, tree *states.Tree) (cid.Cid, error) {
```

Comment:
It's a bit strange that this _returns_ the new miner's state (without updating the state tree) but updates the power actor in the state tree.